src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1526,11 +1526,16 @@ def iter_patches(self, templated_file: TemplatedFile) -> Iterator[FixPatch]:
         else:
             # This segment isn't a literal, but has changed, we need to go deeper.
 
+            # If there's an end of file segment or indent, ignore it.
+            segments = self.segments
+            if self.segments[-1].is_type("end_of_file", "indent"):
+                segments = segments[:-1]
+
```

Comment:
Not sure I fully understand, but does this risk having different Trees before and after fixes?

ala we fix one rule, it deletes the indent (without reason)
next rule picks up on a missing indent
produces a fix that it didn't need to

Reply:
Also could this ever produce an empty segments?

`assert segments, 'Segments Cannot be empty'`

Reply:
Ah - this doesn't _mutate_ segments, it just creates a temporary version of `self.segments` that it used for patch iteration. The actual child segments remain unaltered.

The purpose of this is to ensure that an insertion at the end of a file gets treated properly using the `"end_point"` code path rather than the presence of the new `end_of_file` marker forcing it down the `"mid_point"` path which leads to bugs.

If it results in an empty `segments` list then that won't cause any issues here. We just won't iterate through anything.

Reply:
I've added a note to explain that more clearly.